src/tokens/AssetBalanceFiat.tsx

```diff
@@ -0,0 +1,68 @@
+import BigNumber from 'bignumber.js'
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { LayoutChangeEvent, StyleSheet, Text, View } from 'react-native'
+import { useSelector } from 'react-redux'
+import { useDollarsToLocalAmount } from 'src/localCurrency/hooks'
+import { getLocalCurrencySymbol } from 'src/localCurrency/selectors'
+import { positionsStatusSelector, totalPositionsBalanceUsdSelector } from 'src/positions/selectors'
+import Colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import {
+  stalePriceSelector,
+  tokenFetchErrorSelector,
+  totalTokenBalanceSelector,
+} from 'src/tokens/selectors'
+
+interface Props {
+  onLayout?: (event: LayoutChangeEvent) => void
+}
+
+const AssetBalanceFiat = ({ onLayout }: Props) => {
+  const { t } = useTranslation()
+
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol)
+  const totalTokenBalanceLocal = useSelector(totalTokenBalanceSelector)
+  const totalPositionsBalanceUsd = useSelector(totalPositionsBalanceUsdSelector)
+  const totalPositionsBalanceLocal = useDollarsToLocalAmount(totalPositionsBalanceUsd)
+
+  const tokenFetchError = useSelector(tokenFetchErrorSelector)
+  const tokensAreStale = useSelector(stalePriceSelector)
+  const positionsStatus = useSelector(positionsStatusSelector)
+
+  const totalBalanceLocal =
+    totalTokenBalanceLocal || totalPositionsBalanceLocal
+      ? new BigNumber(totalTokenBalanceLocal ?? 0).plus(totalPositionsBalanceLocal ?? 0)
+      : undefined
+
+  const balance = totalBalanceLocal?.toFormat(2) ?? new BigNumber(0).toFormat(2)
+  // TODO handle when fetching asset balances
+  const displayedBalance =
+    tokenFetchError || tokensAreStale || positionsStatus === 'error' ? ' –' : balance
```

Comment:
Hey, not related directly to your feature, but I'm a bit confused about using `tokensAreStale`.
I saw we were already having this logic... But, this means that if any token is stale we are not showing the total balance, right? Even if it's a token that the user doesn't have?
I'm confused because I think we have a bunch of stale tokens in the list, but I'm not seeing this logic applied.

Reply:
Oh, I think I know where the confusion is coming from...
I think the `stalePriceSelector` always returns `false`, because we are filtering the stale tokens [here](https://github.com/valora-inc/wallet/blob/fd340191e4bf884ad884e2ffad29f2713d50c101/src/tokens/selectors.ts#L88).
I'm wondering what we should do about this... It seems that we wanted to avoid showing the balance if we can't calculate the usd price of any of the tokens the user is holding... I think we are not doing that...
Since it's not related to your PR, I guess we can create a debt tech task for this.